### DISCUSSION TITLE
Question about DFEM Method and Reaction Reigion Overflow Issue

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#26497] [Q&A General] [answered]
Hello everyone,

I am currently employing the DFEM method to solve a problem involving multiple materials, each corresponding to a distinct block. While certain reaction types are exclusive to specific materials, I allocate reaction cross-sections for all reaction types to each material, setting the cross-section value to 0 for non-occurring reaction types.

An overflow issue in the output has arisen, as illustrated in the attached image. The image illustrates the occurrence region of a specific reaction type (highlighted in red), which appears larger than the theoretical region (covered in green). 
![Screenshot from 2024-01-07 23-17-51](https://github.com/idaholab/moose/assets/88585434/43eca906-fed9-4aa2-beac-5a880fe37489)
Given that I am using AuxVariables to track reaction rates, and all operations in AuxKernels are based on quadrature points (_qp), I suspect the issue arises from the presence of quadrature points at the interfaces between different materials. These quadrature points belong to adjacent cells and possess properties of one adjacent material each. Although this doesn't impact the program's calculations, moose will merge data for quadrature points at the same location during output, resulting in non-zero reaction rates for adjacent cells.

I am using MONOMIAL variables of the first order, and I am unsure about the accuracy of my speculation. If it is, could you please provide guidance on how to address this situation? 

Appreciate your assistance.

hityyds

**Comment by GiudGiud** [comment, marked as answer]
Hello

This is an output problem I think. Higher order (discontinuous) elemental variables are mostly unsupported in exodus & visualization, as is discussed here https://github.com/idaholab/moose/issues/8635 (due to a string of issues, this issue dates but we have not addressed all of them)

You are visualizing a first MONOMIAL. Could you drop down to visualizing a constant monomial instead? This is well supported. 

Guillaume

**Reply by hityyds** [reply]
I followed your advice and used the CONSTANT variable to output the result, and indeed the overflow problem disappeared. However, may I ask, is this really a output problem? Have you previously attempted to test similar issues?

**Reply by lindsayad** [reply]
This is a common output problem. @YaqiWang might weigh in with his experience/suggestions for visualizing non constant discontinuous functions

**Reply by hityyds** [reply]
I get it. Thank you for your reply.
